src/onCLS.ts

```diff
@@ -62,9 +65,52 @@ export const onCLS = (onReport: CLSReportCallback, opts?: ReportOpts) => {
       let sessionValue = 0;
       let sessionEntries: PerformanceEntry[] = [];
 
+      const initNewCLSMetric = (navigation?: Metric['navigationType']) => {
+        metric = initMetric('CLS', 0, navigation);
+        report = bindReporter(
+          onReport,
+          metric,
+          thresholds,
+          opts!.reportAllChanges
+        );
+        sessionValue = 0;
+        sessionEntries = [];
+      };
+
       // const handleEntries = (entries: Metric['entries']) => {
       const handleEntries = (entries: LayoutShift[]) => {
+        let pageUrl = '';
```

Comment:
This would compute url every time new shifts are reported (~once per animation frame).

I think you can cache it until a navigation happens.

Reply:
Given how easy it is to look up the pageUrl, I've decoded to simplify this and go back to reporting only he navId. So this is now gone.